### PR TITLE
Create Search screen

### PR DESCRIPTION
**Pull request by vasileGrad**

### CODE REVIEW

File: app/src/main/java/com/example/hungrywolfs/adapters/MealGridAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MealGridViewHolder {
+        return MealGridViewHolder(
+            GridViewItemBinding.inflate(LayoutInflater.from(parent.context))
+        )
+    }
+
+    override fun onBindViewHolder(holder: MealGridViewHolder, position: Int) {
+        val mealInfo = getItem(position)
```
Comment (cristiwd) [inline app/src/main/java/com/example/hungrywolfs/adapters/MealGridAdapter.kt]: No need to create another val, you could use **getItem(position)** as parameter for bind method.

File: app/src/main/java/com/example/hungrywolfs/adapters/BindingAdapters.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+@BindingAdapter("strMeal")
+fun bindText(textView: TextView, text: String?) {
+    textView.text = text
+}
+
+@BindingAdapter("listMeals")
+fun RecyclerView.bindRecycler(listMeals: List<MealInfo>?) {
+    val adapter = this.adapter as MealGridAdapter
```
Comment (cristiwd) [inline app/src/main/java/com/example/hungrywolfs/adapters/BindingAdapters.kt]: An improvement here would be to use [as?](https://kotlinlang.org/docs/typecasts.html#safe-nullable-cast-operator) and then use the adapter as **adapter?**...

File: app/src/main/java/com/example/hungrywolfs/adapters/BindingAdapters.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        val imgUri = imgUrl.toUri()
+        imgView.load(imgUri) {
+            placeholder(R.drawable.loading_animation)
+            error(R.drawable.ic_broken_image)
+        }
+    }
+}
+
+@BindingAdapter("strMeal")
+fun bindText(textView: TextView, text: String?) {
```
Comment (cristiwd) [inline app/src/main/java/com/example/hungrywolfs/adapters/BindingAdapters.kt]: An improvement for these 2 adapters is to remove the first parameter and actually create an [extension function](https://kotlinlang.org/docs/extensions.html#extension-functions) like you created for the RecyclerView below.

File: app/src/main/java/com/example/hungrywolfs/adapters/BindingAdapters.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import androidx.core.net.toUri
+import androidx.databinding.BindingAdapter
+import androidx.recyclerview.widget.RecyclerView
+import coil.load
+import com.example.hungrywolfs.R
+import com.example.hungrywolfs.network.MealInfo
+
+@BindingAdapter("strMealThumb")
+fun bindCircleImage(imgView: ImageView, imgUrl: String?) {
+    imgUrl?.let {
```
Comment (cristiwd) [inline app/src/main/java/com/example/hungrywolfs/adapters/BindingAdapters.kt]: SMALL suggestion. To reduce the indentation of this method, you could remove the imgUrl?.let block and use the **imgUrl ?: return** as the first line of the method.

**Review by cristiwd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/hungrywolfs/adapters/BindingAdapters.kt (hunk 2 above)
Comment (vasileGrad) [reply app/src/main/java/com/example/hungrywolfs/adapters/BindingAdapters.kt]: Thanks. The as? allows the cast to return null when the left-hand side is null

**Review by cristiwd — APPROVED** [review]

**Pull request merged**